loopchain/blockchain/blocks/v0_3/block_builder.py

```diff
@@ -129,20 +130,27 @@ def _build_receipts_hash(self):
         return block_prover.get_proof_root()
 
     def build_reps_hash(self):
-        try:
-            if self.reps_hash is not None:
-                return self.reps_hash
-
-            self.reps_hash = self._build_reps_hash()
+        if self.reps_hash is not None:
             return self.reps_hash
-        finally:
-            if self.next_reps_hash is None:
-                self.next_reps_hash = self.reps_hash
+
+        self.reps_hash = self._build_reps_hash()
+        return self.reps_hash
 
     def _build_reps_hash(self):
         block_prover = BlockProver((rep.extend() for rep in self.reps), BlockProverType.Rep)
         return block_prover.get_proof_root()
 
+    def build_next_reps_hash(self):
+        if self.next_reps_hash is not None:
+            return self.next_reps_hash
+
+        self.next_reps_hash = self._build_next_reps_hash()
+        return self.next_reps_hash
+
+    def _build_next_reps_hash(self):
+        block_prover = BlockProver((rep.extend() for rep in self.next_reps), BlockProverType.Rep)
```

Comment:
How about checking `self.reps == self.next_reps` first, then use `self.reps_hash` instead, if two reps are identical?

I'm not sure but it might be faster in certain situations.

> get root hash of all leaves ALWAYS VS compare all items of sequence

Reply:
I almost agree with you, most cases are the same maybe. But `reps` must be built earlier than `next_reps` always if do as your opinion. I worry about the dependency of the sequence between them. Let me know your opinion.

Reply:
You're right.

My suggestion above is assured IF the reps hash is built BEFORE next reps hash.

Please keep it on.